File: java/src/com/zhy/DesignPattern/Adapter/CalculationClassAdapter.java
package com.zhy.DesignPattern.Adapter;

public class CalculationClassAdapter extends CalculationAdaptee implements ICalculation {

	@Override
	public double Add(double x, double y) {

		int value = super.AddMethod((int)x, (int)y);
        return (double)value;
	}
}

File: csharp/AdapterPattern/CalculationClassAdapter.cs

namespace AdapterPattern
{
    public class CalculationClassAdapter : CalculationAdaptee, ICalculation
    {
        public double Add(double x, double y)
        {
            int value = base.AddMethod((int)x, (int)y);
            return (double)value;
        }
    }
}
